src/openms/include/OpenMS/FORMAT/FileHandler.h

```diff
@@ -121,7 +122,7 @@ namespace OpenMS
       @exception Exception::ParseError is thrown if an error occurs during parsing
     */
     template <class PeakType>
-    bool loadExperiment(const String& filename, MSExperiment<PeakType>& exp, FileTypes::Type force_type = FileTypes::UNKNOWN, ProgressLogger::LogType log = ProgressLogger::NONE, const bool compute_hash = true)
+    bool loadExperiment(const String& filename, MSExperiment<PeakType>& exp, FileTypes::Type force_type = FileTypes::UNKNOWN, ProgressLogger::LogType log = ProgressLogger::NONE, const bool rewrite_source_file = true, const bool compute_hash = true)
```

Comment:
Yes that is true. I did it kind of intentional because of the dependency between rewrite_source_file and recompute hash. I could change it back but this would make the API a bit more awkward. I think it is so far only used internally so I didn't have a problem with changing it. What I would actually like to do is to change the default behavior to not rewrite the source files. It is more correct and over/rewriting should be only used in very special cases.